src/workerd/api/node/util.c++

```diff
@@ -240,4 +241,39 @@ jsg::JsValue UtilModule::getBuiltinModule(jsg::Lock& js, kj::String specifier) {
   return js.undefined();
 }
 
+void UtilModule::processExitImpl(jsg::Lock& js, int code) {
+  if (IoContext::hasCurrent()) {
+    // There are a few things happening here. First, we abort the current IoContext
+    // in order to shut down this specific request....
+    auto message =
+        kj::str("The Node.js process.exit(", code, ") API was called. Canceling the request.");
+    auto& ioContext = IoContext::current();
+    // If we have a tail worker, let's report the error.
+    KJ_IF_SOME(tracer, ioContext.getWorkerTracer()) {
+      auto ex = KJ_ASSERT_NONNULL(js.error(message).tryCast<jsg::JsObject>());
+      tracer.addException(ioContext.now(), ex.get(js, "name"_kj).toString(js),
+          ex.get(js, "message"_kj).toString(js), ex.get(js, "stack"_kj).toString(js));
+      ioContext.abort(js.exceptionToKj(ex));
```

Comment:
Why use a different exception here vs. the non-traced case? Seems weird for tracing to change behavior?

Reply:
Yeah this felt a bit funky but the reason (admittedly not very good) is that `tracer.addException(...)` is really set up for us to extract the `name`, `message`, and `stack` (most importantly the stack)... the stack being the part that is most important to send to the tail worker. I also wanted to avoid the additional overhead of creating the JS error in the case tracing is not being used. I don't feel strongly about how any of this bit is implemented tho.

Reply:
Added a comment.